ocs_ci/ocs/resources/bucketclass.py

```diff
@@ -0,0 +1,123 @@
+from botocore.exceptions import ClientError
+
+from ocs_ci.ocs.ocp import OCP
+import logging
+from ocs_ci.framework import config
+from tests import helpers
+
+log = logging.getLogger(__name__)
+
+
+class BucketClass():
+    """
+    A class that represents BucketClass objects
+
+    """
+    def __init__(self, name, backingstores, placement):
+        self.name = name
+        self.backingstores = backingstores
+        self.placement = placement
+
+    # TODO: verify health of bucketclass
+
+    def delete(self):
+        log.info(f'Cleaning up bucket class {self.name}')
+
+        OCP(
+            namespace=config.ENV_DATA['cluster_namespace']
+        ).exec_oc_cmd(
+            command=f'delete bucketclass {self.name}',
+            out_yaml_format=False
+        )
+
+        log.info(
+            f"Verifying whether bucket class {self.name} exists after deletion"
+        )
+        # Todo: implement deletion assertion
+
+
+def bucket_class_factory(request, mcg_obj, backingstore_factory):
+    """
+    Create a bucket class factory. Calling this fixture creates a new custom bucket class.
+    For a custom backingstore(s), provide the 'backingstore_dict' parameter.
+
+    Args:
+        request (object): Pytest built-in fixture
+        mcg_obj (MCG): An MCG object containing the MCG S3 connection credentials
+        backingstore_factory: Factory for backing store creation
+
+    """
+    interfaces = {
+        'oc': mcg_obj.oc_create_bucketclass,
+        'cli': mcg_obj.cli_create_bucketclass
+    }
+    created_bucket_classes = []
+
+    def _create_bucket_class(bucket_class_dict):
+        """
+        Creates and deletes all bucket classes that were created as part of the test
+
+        Args:
+            bucket_class_dict (dict): Dictionary containing the description of the bucket class.
+                possible keys and values are:
+                - interface (str): The interface to use for creation of buckets.
+                    OC | CLI
+                - backingstore_dict (dict/list): Either existing list of backingstores or
+                    A dictionary compatible with the backing store factory requirements.
+                - placement (str): The Placement policy for this bucket class.
+                    Spread | Mirror
+                if no key is provided default values will apply.
+
+        Returns:
+            list: A Bucket Class object.
+
+        """
+        if 'interface' in bucket_class_dict:
```

Comment:
why not passing `interface` as an explicit arg to  `_create_bucket_class()`?

Reply:
its a possibility but since we already used the same kind of format for the other backing store and uls factories i thought it would be more consistent to accept a single dictionary argument